apps/aecore/priv/exometer_predefined.script

```diff
@@ -1,29 +1,40 @@
- {[ae,epoch,aecore,peers,ping,success]       , counter     , []},
- {[ae,epoch,aecore,peers,ping,failure]       , counter     , []},
- {[ae,epoch,aecore,tx_pool,push]             , counter     , []},
- {[ae,epoch,aecore,tx_pool,push,illegal]     , counter     , []},
- {[ae,epoch,aecore,tx_pool,push,error]       , counter     , []},
- {[ae,epoch,aecore,tx_pool,push,ignore]      , counter     , []},
- {[ae,epoch,aecore,tx_pool,gced]             , counter     , []},
- {[ae,epoch,aecore,tx_pool,origin_gced]      , counter     , []},
- {[ae,epoch,aecore,sync,progress]            , gauge       , []}
+%% for information about how to create, update and read metrics.
+[ {[ae,epoch,aecore,chain,height]                      , gauge       , []}
+, {[ae,epoch,aecore,chain,top_change,interval]         , histogram   , []}
+, {[ae,epoch,aecore,mining,blocks_mined]               , counter     , []}
+, {[ae,epoch,aecore,mining,interval]                   , histogram   , []}
+, {[ae,epoch,aecore,mining,retries]                    , counter     , []}
+, {[ae,epoch,aecore,peers,blocked]                     , gauge       , []}
+, {[ae,epoch,aecore,peers,count]                       , gauge       , []}
+, {[ae,epoch,aecore,peers,errored]                     , gauge       , []}
+, {[ae,epoch,aecore,peers,inbound]                     , gauge       , []}
+, {[ae,epoch,aecore,peers,outbound]                    , gauge       , []}
+, {[ae,epoch,aecore,peers,ping,failure]                , counter     , []}
+, {[ae,epoch,aecore,peers,ping,success]                , counter     , []}
+, {[ae,epoch,aecore,peers,unverified]                  , gauge       , []}
+, {[ae,epoch,aecore,peers,verified]                    , gauge       , []}
+, {[ae,epoch,aecore,sync,progress]                     , gauge       , []}
+, {[ae,epoch,aecore,tx_pool,gced]                      , counter     , []}
+, {[ae,epoch,aecore,tx_pool,origin_gced]               , counter     , []}
+, {[ae,epoch,aecore,tx_pool,push,error]                , counter     , []}
+, {[ae,epoch,aecore,tx_pool,push,ignore]               , counter     , []}
+, {[ae,epoch,aecore,tx_pool,push,illegal]              , counter     , []}
+, {[ae,epoch,aecore,tx_pool,push]                      , counter     , []}
+
+%% The execution time metrics are captured in micro-seconds.
+
+%% Keep 10 minutes of data, ~= 200 values, with a new micro-block every 3 seconds.
+, {[ae,epoch,aecore,blocks,micro,txs_execution_time,success]    , histogram , [{time_span, 10 * 60 * 1000}]}
+, {[ae,epoch,aecore,blocks,micro,txs_execution_time,error]      , histogram , [{time_span, 10 * 60 * 1000}]}
+, {[ae,epoch,aecore,blocks,micro,insert_execution_time,success] , histogram , [{time_span, 10 * 60 * 1000}]}
+, {[ae,epoch,aecore,blocks,micro,insert_execution_time,error]   , histogram , [{time_span, 10 * 60 * 1000}]}
+
+%% Keep 4 hours of data, ~= 120 values, with a new key-block every 2 minutes.
```

Comment:
It could be metrics related, but we expect key-blocks every 3 minutes

Reply:
Right, I tried to be conservative here. This way we don't get too many values into the histogram to make computation slow. If its every 3 minutes with these settings there will be 80 entries which is close enough to 100 to be useful.